src/mococrw/hsm.h

```diff
@@ -26,17 +26,49 @@ class ECCSpec;
 class RSASpec;
 
 /**
- * This struct currently contains PKCS#11 attributes which are changeable on key creation.
+ * This class currently contains PKCS#11 attributes which are changeable on key creation.
  * In the future also parameters for other keystorage interfaces can be added.
  */
-struct HsmKeyParams
+class HsmKeyParams
 {
+public:
+    class Builder;
+
+    bool isExtractable() const { return cka_extractable; }
```

Comment:
Do we really want to expose PKCS#11 slang here? Wouldn't it be better if this provides some Abstraction and we say here that the param is that the is is extractable or not? Sensitive is really not self-explanatory.